extensions/cornerstone/src/commandsModule.js

```diff
@@ -157,9 +157,149 @@ const commandsModule = ({ servicesManager }) => {
         },
       });
     },
+    updateTableWithNewMeasurementData(measurementData) {
+      const {
+        toolType,
+        measurementNumber,
+        location,
+        description,
+      } = measurementData;
+
+      // Update all measurements by measurement number
+      const measurementApi = OHIF.measurements.MeasurementApi.Instance;
+      const measurements = measurementApi.tools[toolType].filter(
+        m => m.measurementNumber === measurementNumber
+      );
+
+      measurements.forEach(measurement => {
+        measurement.location = location;
+        measurement.description = description;
+
+        measurementApi.updateMeasurement(measurement.toolType, measurement);
+      });
+
+      measurementApi.syncMeasurementsAndToolData();
+
+      // Update images in all active viewports
+      cornerstone.getEnabledElements().forEach(enabledElement => {
+        cornerstone.updateImage(enabledElement.element);
+      });
+    },
+    showContextMenu({ event }) {
+      const { UIContextMenuService } = servicesManager.services;
+      UIContextMenuService.show({ event });
+    },
+    hideContextMenu() {
+      const { UIContextMenuService } = servicesManager.services;
+      UIContextMenuService.hide();
+    },
+    showLabellingFlow({ defaultPosition, centralize, props }) {
+      const { UILabellingFlowService } = servicesManager.services;
+      UILabellingFlowService.show({
+        defaultPosition,
+        centralize,
+        props,
+      });
```

Comment:
If this is just a generic command that has no cornerstone dependency and is a pass-through call to a service, why can't we just call the service directly? This feels like an unnecessary layer?

Reply:
Done! thanks.